flink-libraries/flink-table/src/test/scala/org/apache/flink/api/scala/batch/sql/SelectITCase.scala

```diff
@@ -146,4 +148,21 @@ class SelectITCase(
     tEnv.sql(sqlQuery)
   }
 
+  @Test
+  def testConstantReduce(): Unit = {
```

Comment:
I think the assertion of this this test should not check the name of the generate Flink operator.
Instead I propose the following:
- split the `translate()` method into an `optimize()` method that generates the optimized `RelNode` tree and a `translate()` method that translates into a `DataSet` / `DataStream` program.
- make the `optimize()` method `private[flink]` and therefore accessible from a unit test
- add a `BatchTableEnvironmentTest` and a `StreamTableEnvironmentTest` which check that the optimized `RelNode` tree contains reduced expressions.